### DISCUSSION TITLE
Ideas for Future Development of Boa.

### DISCUSSION THREAD

**Discussion by ianBBB** [ianBBB/boa-constructor#12] [Ideas]
During the course of working on Boa, ideas float in and out of my mind; 
"wouldn't it be good if . . . " 
"It should be simpler to do this . ."
"Why on Earth did they do it like **that** ?"
Missing feature and various improvements; they need to be captured. 
So here is a place to store them.
I may one day get around to implementing them.

**Comment by ianBBB** [comment]
### Add drag-and-drop to Designer mode.
In line with current expectations of GUI operation, adding drag and drop would seem like a good idea. Still need pick-with-tick for developing full-screen applications.

**Comment by ianBBB** [comment]
### A Better Trace System/Function (Please!)
Yes, the original one was primitive. I tried to re-hash it but botched it up. I note that there is **_trace_** module in Python so I hope to use this to deliver a better debugging tool.

**Comment by bleeisme** [comment]
I would like / suggest:
1. git integration
2. Environment awareness (prefer Conda) (launch code in target environment)
3. Making projects files open as a child of the project parent (currently the only hierarchy is that child file tabs open to the right of the project tab (I keep many projects open at same time)
4. More flexible dynamic table creation:  For example 10 pairs of buttons & static text laid out using the designer
5. Designer support for derived types (either full support, or just a placeholder in designer view)

BTW - I am pumped that I found your work.  I started using boa-constructor close to 20 years ago.

**Reply by ianBBB** [reply]
Sorry, I missed responding to this here.
Thank you for these ideas. At this point, most are well beyond my skillset but they needed to be said/recorded.
The business of just getting what used to work, working again is filling my time at this stage and there is still a long way to go to achieve that. I can certainly relate to the 10 pairs of StaticText and Buttons issue and this is probably the first one I will look at but please do not hold your breathe.

**Comment by ianBBB** [comment]
Need to re-do the companion for the style of ListView/Listctrl.
At the moment, it opens all the options as tick boxes. The problem with this is that some of the options are mutually exclusive (like Headers, and No Headers). In particular, there are five mode setting and only exactly one is allowed to be selected.

**Comment by ianBBB** [comment]
Suggest the implementation of the HTML2 control (from wxPython). The original Boa Constructor supported an early HTML. Today HTML supports CSS and Javascript and the HTML2 in wxPython is the control that renders these HTML pages. As internet access is so important to modern applications, it seems important to provide this contrrol.
